qiime2/sdk/plugin_manager.py

```diff
@@ -93,18 +93,38 @@ def _integrate_plugin(self, plugin):
     # databases rollback/commit
 
     @property
+    def importable_formats(self):
+        """Return set of formats that are importable.
+
+        A format is importable in a QIIME 2 deployment if it can be transformed
+        into at least one of the canonical semantic type formats.
+
+        """
+        importable_formats = {}
+        for name, record in self.formats.items():
+            for type_format in self.type_formats:
+                from_type = transform.ModelType.from_view_type(
```

Comment:
Sorry I didn't notice this before: can you move the `from_type` instantiation out of the inner loop since it doesn't change in this loop? That may save a little time.